test/manipulation.test.js

```diff
@@ -503,6 +503,25 @@ describe('manipulation', function () {
         });
     });
 
+    it('should allow same stringified id value on updateAttributes',
+      function(done) {
+        var pid = person.id;
+        if (typeof person.id === 'object' || typeof person.id === 'number') {
+          // For example MongoDB ObjectId
+          pid = person.id.toString();
+        }
+        person.updateAttributes({id: pid, name: 'John'},
+          function(err, p) {
+            should.not.exist(err);
```

Comment:
Please use `if (err) return done(err)`, it gives better failure messages in the test output. Same comment applies to L517.